### PR TITLE
Refactor `alfa-selector`

### PR DESCRIPTION
**Pull request by Jym77**
Break down the huge mono-file package into separate files.

### CODE REVIEW

**Comment by Jym77** [comment]
!pr extract

**Review by rcj-siteimprove — COMMENTED** [review]
Looks good, just a few comments.

File: packages/alfa-selector/src/selector/relative.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+/**
+ * {@link https://drafts.csswg.org/selectors/#typedef-relative-selector}
+ */
+// const parseRelative = map(pair(parseCombinator, parseComplex), (result) => {
+//   const [combinator, selector] = result;
+
+//   return Relative.of(combinator, selector);
+// });
```
[the remark this answers is not in the document]
Comment (Jym77) [reply packages/alfa-selector/src/selector/relative.ts:93]: This was already commented out in the current code. Didn't feel like removing it completely 🤷

**Review by rcj-siteimprove — APPROVED** [review]

**Review by rcj-siteimprove — APPROVED** [review]

**Comment by Jym77** [comment]
!pr extract

**Review by rcj-siteimprove — APPROVED** [review]

**Pull request merged**